src/tests/ftest/util/daos_io_conf.py

```diff
@@ -0,0 +1,112 @@
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+import os
+from avocado.utils import process
+
+from apricot import TestWithServers
+from command_utils import ExecutableCommand, CommandFailure, FormattedParameter
+from command_utils import BasicParameter
+from test_utils import TestPool
+
+class IoConfGen(ExecutableCommand):
+    """Defines an object for the daos_gen_io_conf and daos_run_io_conf commands.
+
+    :avocado: recursive
+    """
+    def __init__(self, path="", env=None):
+        """Create a ExecutableCommand object.
+
+        Uses Avocado's utils.process module to run a command str provided.
+
+        Args:
+            command (str): string of the command to be executed.
+            path (str, optional): path to location of command binary file.
+                Defaults to ""
+        """
+        super(IoConfGen, self).__init__("/run/run_gen_io_conf/*",
+                                        "daos_gen_io_conf")
+        self.verbose = True
+        self.env = env
+        self.ranks = FormattedParameter("-g {}")
+        self.targets = FormattedParameter("-t {}")
+        self.obj_num = FormattedParameter("-o {}")
+        self.akeys = FormattedParameter("-a {}")
+        self.dkeys = FormattedParameter("-d {}")
+        self.re_size = FormattedParameter("-s {}")
+        self.obj_class = FormattedParameter("-O {}")
+        self.filename = BasicParameter(None, "testfile")
+
+    def run_conf(self):
+        """Run the daos_run_io_conf command as a foreground process.
+
+        Raises:
+            CommandFailure: if there is an error running the command
+
+        """
+        command = " ".join([os.path.join(self._path, "daos_run_io_conf"),
+                            self.filename.value])
+        kwargs = {
+            "cmd": command,
+            "timeout": self.timeout,
+            "verbose": self.verbose,
+            "allow_output_check": "combined",
+            "shell": True,
+            "env": self.env,
+            "sudo": self.sudo,
+        }
+        try:
+            # Block until the command is complete or times out
+            return process.run(**kwargs)
+
+        except process.CmdError as error:
+            # Command failed or possibly timed out
+            msg = "Error occurred running '{}': {}".format(command, error)
+            self.log.error(msg)
+            raise CommandFailure(msg)
+
+class IoConfTestBase(TestWithServers):
+    """Base rebuild test class.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a IoConfTestBase object."""
+        super(IoConfTestBase, self).__init__(*args, **kwargs)
```

Comment:
It looks like this definition really isn't needed.  It would only be required if the `IoConfTestBase` class needs to perform some addition actions currently not performed by the `TestWithServers.__init__()`.  Are there any extra steps required in the `__init__()` that are not specified here?  

If there was a need to run `daos_gen_io_conf` in one method and then `daos_run_io_conf` in a separate class method, I could see needing a `self.io_conf = None` statement included here, in which case it may make sense to have a `def setup_io_conf(self)` method.  Of course this could all be implemented if needed in the future.